File: PYTHON/01/arrays_medium_problems/longest_consecutive_sequence.py
class Solution:
    def function(self,arr):
        if len(arr) == 0: return 0
        arr = set(arr)
        longest = 1
        for num in arr:
            if num-1 not in arr:
                cnt = 1
                while num+1 in arr:
                    num += 1
                    cnt +=1
                longest = max(longest, cnt)
        return longest

if __name__ == '__main__':
    arr = [100, 4, 200, 1, 3, 2]
    obj = Solution()
    result = obj.function(arr)
    print(result)

File: JAVA/src/arrayMedium/longestConsecutiveSequence.java
package arrayMedium;

import java.util.HashSet;

public class longestConsecutiveSequence {
    public static int function(int[] arr){
        HashSet<Integer> set = new HashSet<>();
        if(arr.length == 0) return 0;
        int longest = 1;
        for (int i = 0; i < arr.length; i++) {
            set.add(arr[i]);
        }
        for (int i = 0; i < arr.length; i++) {
            if(!set.contains(arr[i]-1)){
                int cnt = 1;
                int x = arr[i];
                while (set.contains(x+1)){
                    x += 1;
                    cnt +=1;
                }
                longest = Math.max(longest,cnt);
            }
        }
        return longest;
    }
    public static void main(String[] args) {
        int[] arr  = {100, 4, 200, 1, 3, 2};
        int result = function(arr);
        System.out.println(result);
    }
}
